core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -277,8 +277,8 @@ private[spark] class RestSubmissionClient(master: String) extends Logging {
   private def getBaseUrl(master: String): String = {
     var masterUrl = master
     supportedMasterPrefixes.foreach { prefix =>
-      if (master.startsWith(prefix)) {
-        masterUrl = master.stripPrefix(prefix)
+      if (masterUrl.startsWith(prefix)) {
+        masterUrl = masterUrl.stripPrefix(prefix)
```

Comment:
So the problem is that the master URL is being prepended with spark://, so mesos://1.1.1.1 becomes spark://mesos://1.1.1.1, and therefore just stripping it once doesn't work. While driving back home I realize this fix is not the right fix, I should actually fix where the append is happening which is up where master variable is being assigned. I'll create a new pr and jira against master.

How do I then get a or against master into 1.4 branch?